internal/acctest/acctest.go

```diff
@@ -1,21 +1,28 @@
 package acctest
 
 import (
+	"context"
+	"log"
 	"os"
 	"testing"
 
 	"github.com/elastic/terraform-provider-elasticstack/provider"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-var Providers map[string]func() (*schema.Provider, error)
+var Providers map[string]func() (tfprotov5.ProviderServer, error)
 var Provider *schema.Provider
```

Comment:
This variable is almost always just used to grab an Elasticsearch client and should be removed IMO. Calling it out here since it removes the need to return the `primaryProvider` from the server factory.